### DISCUSSION TITLE
Pyaedt v0.5.0 Released with Linux CPython Support

### DISCUSSION THREAD

**Discussion by maxcapodi78** [ansys/pyaedt#1366] [Announcements]
I’d like to inform you about the following update: I’ve just released Pyaedt v0.5.0. This contains a major update especially for linux users.
•	AEDT 2022R2 and pyaedt v0.5.0 works in CPython Linux (3.7-3.10) without any limitation like in windows. No need anymore to run with Ironpython
•	Edb work well in Linux CPython without any limitation. No need anymore to run with Ironpython
•	Full support of Python 3.10   
•	Drop of support of Python 3.6 and lower (except Ironpython)    
•	Pyaedt supports new AEDT GRPC Api. It means that a user can establish a remote connection with an existing remote AEDT session listening on a specified port and run it like locally (not valid for EDB).

In order to use Pyaedt in Cpython in Linux you need:

•	CPython 3.7-3.10
•	AEDT 22R2 installed. It has been released on July 7th 2022 on Customer Portal.
•	Setup the following environment variables:
o	export ANSYSEM_ROOT222=/path/to/AnsysEM/v222/Linux64
o	export LD_LIBRARY_PATH=$ANSYSEM_ROOT222/common/mono/Linux64/lib:$ANSYSEM_ROOT222/Delcross:$LD_LIBRARY_PATH

After that you can open a python console and run

# To launch Edb and open my_file. 
`edbversion` argument is set by default to ANSYSEM_ROOT222.

```
from pyaedt import Edb
edb = Edb("my_file")
```

# Import one of AEDT Application and start a new session on a randomic grpc port or to connect to existing grpc session.

```
from pyaedt import Maxwell3d
m3d = Maxwell3d()   
```

# To start new session on specified port or to connect to exsting grpc session on portnumber. 
`non_graphical` option can be used.

```
m3d = Maxwell3d(port=portnumber)  
```
# To connect to an existing AEDT session listening on port portnumber in a remote machine "machine name.
The AEDT session has to be already open and listening
To start the aedt session on remote machine : /path/to/AnsysEM/v222/Linux64/ansysedt -grpcsrv portunumber
This approach is working also cross platform (Windows-Windows, Windows-Linux, Linux-Windows)
```
m3d = Maxwell3d(machine="machinename", port=portnumber)
```
![image](https://user-images.githubusercontent.com/77293250/177941940-94f17b23-eab5-4c95-837c-4b6d5b20d6a4.png)

Pyaedt Team

**Comment by schnib** [comment]
Hi,
is it possible to run pyaedt through a WSL python interpreter, but using the Windows ANSYS AEDT installation, using this approach?
What would be the correct path for LD_LIBRARY_PATH if it's possible?
Thanks

**Reply by maxcapodi78** [reply]
ideally yes. I've never tried

**Reply by MaxJPRey** [reply]
@schnib Can you share more info about why you would use WSL over the command prompt if you are operating on a Windows machine and wants to use the Windows installation of AEDT. It seems like adding an extra layer/proxy at first sight.
Thanks

**Reply by schnib** [reply]
It's mostly convenience. I am more familiar with developing in a Linux environment, but have to use the Windows AEDT version at the moment.
If it's not possible without any major workarounds, it's not really more convenient and I'd stick to developing in the Windows environment for now. 
Thanks

**Reply by Woogachaka** [reply]
I am also interested in this approach, does anyone know if there's a way to use a windows installation of AEDT with a WSL environment? My context is a graduate student lab where we have much more flexibility in our python usage and development in WSL but only have the windows version of AEDT available to us.
